File: src/main/java/leetcode/hot100/LeetCode461.java
package leetcode.hot100;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/19 14:11
 */
public class LeetCode461 {

    public int hammingDistance(int x, int y) {
        int s = x ^ y, ret = 0;
        while (s != 0) {
            ret += s & 1;
            s >>= 1;
        }
        return ret;

    }
}

File: src/main/go/leetcode/lists/hot100/LeetCode461.go
package hot100

//https://www.youtube.com/watch?v=i4gGVvhmLds
func hammingDistance(x int, y int) int {
	s := x ^ y
	ret := 0
	for s != 0 {
		ret += s & 1
		s >>= 1
	}
	return ret
}
